### DISCUSSION TITLE
Support configuration of WebSocket's "max_queue"

### DISCUSSION THREAD

**Discussion by ssproessig-thales** [Kludex/uvicorn#2032] [Ideas] [answered]
Currently, uvicorn supports configuring WebSocket's `max_size` using `Config.ws_max_size`.

I'd be interested in configuring `max_queue` as well, e.g. via `ws_max_queue`.

I did not find any way to change the built-in default of `2**5=32` that is set for the default value in `WebSocketCommonProtocol.__init__` otherwise.

Would this be an ok idea? Or are there maybe problems?

**Comment by Kludex** [comment]
I don't know. Does `wsproto` also allows that configuration?

**Reply by ssproessig-thales** [reply, marked as answer]
Didn't see any `queue` in their code. Seems like they use generators and `yield` events out directly

- https://github.com/python-hyper/wsproto/blob/main/src/wsproto/frame_protocol.py#L568C28-L568C28
- https://github.com/python-hyper/wsproto/blob/main/src/wsproto/connection.py#L150
- https://github.com/python-hyper/wsproto/blob/main/src/wsproto/frame_protocol.py#L502

Looks like the raw socket buffer is used directly.

I never used `wsproto`, though. 

Would be a `websockets` only option (as `max_size`, isn't it?)...

**Reply by Kludex** [reply]
yep, since we already have `max_size`, then we can have `max_queue` as well.
